File: owlready2/owlready2-example/Evaluation.py
import owlready2
import time

def eval_load_ontology(location):
    start_time = time.time()
    onto = owlready2.get_ontology(location).load(reload=True);
    end_time = time.time()
    print("Loading Time: ", end_time - start_time)
    return end_time - start_time

def perform_loading_evaluation(RUN=5):
      for ontology_name, ontology_file in ontologies_map.items():
        print("Ontology " + ontology_name)
        evaluation_time = 0
        for x in range(RUN):
            evaluation_time += eval_load_ontology(ontology_file)

        print("Everage loadding time of " + ontology_name + ": ", evaluation_time / RUN)
        print("")

def load_ontology(location):
    print("Loading " + location)
    onto = owlready2.get_ontology(location).load(reload=True)
    onto_classes = list(onto.classes())
    onto_individuals = list(onto.individuals())
    print("Number of classes:", len(onto_classes))
    print("Number of individual: ", len(onto_individuals))
    return onto

def perform_pellet_reasoner(ontology):
    print("Loading pellet reasoner")
    with ontology:
        start_time = time.time()
        owlready2.sync_reasoner_pellet(debug=0)
        end_time = time.time()

    print("Run Time: ", end_time - start_time)
    return end_time - start_time


def perform_hermit_reasoner(ontology):
    print("Loading hermit reasoner")
    with ontology:
        start_time = time.time()
        owlready2.sync_reasoner_hermit(debug=0)
        end_time = time.time()

    print("Run Time: ", end_time - start_time)
    return end_time - start_time

def perform_reasoning_evaluation():
    RUN = 5
    for ontology_name, ontology_file in ontologies_map.items():
        print("Ontology " + ontology_name)
        evaluation_time = 0
        for x in range(RUN):
            onto = load_ontology(ontology_file)
            onto.base_iri = ontology_name
            evaluation_time += perform_hermit_reasoner(onto)

        print("Everage runtime of HermiT ", evaluation_time / RUN)
        print("")

        evaluation_time = 0
        for x in range(RUN):
            onto = load_ontology(ontology_file)
            evaluation_time += perform_pellet_reasoner(onto)

        print("Everage runtime of Pellet ", evaluation_time / RUN)
        print("")

if __name__ == '__main__':

    ontologies_map = {
        #"http://vicodi.org/ontology": "vicodi_all.owl",
        #"http://www.ifomis.org/acgt/1.0": "ACGT.owl",
        "http://www.co-ode.org/ontologies/galen": "full-galen.owl",
        "http://purl.org/sig/ont/fma.owl": "fma.owl",
        "http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#":  "ncit.owl",
         "http://purl.bioontology.org/ontology/MESH/": "MESH.owl"

    }

    owlready2.onto_path.append("../../ontologies/")
    owlready2.reasoning.JAVA_MEMORY = 5000

    #perform_loading_evaluation()
    perform_reasoning_evaluation()

File: Jena/jena-owl-example/src/main/java/no/sintef/skytrack/jena/Evaluation.java
package no.sintef.skytrack.jena;

import java.io.FileWriter;
import java.io.IOException;
import java.util.Iterator;
import java.util.LinkedHashMap;
import java.util.Map;

import org.apache.jena.ontology.OntDocumentManager;
import org.apache.jena.ontology.OntModel;
import org.apache.jena.ontology.OntModelSpec;
import org.apache.jena.rdf.model.InfModel;
import org.apache.jena.rdf.model.ModelFactory;
import org.apache.jena.reasoner.Reasoner;
import org.apache.jena.reasoner.ReasonerRegistry;
import org.apache.jena.reasoner.ValidityReport;
import org.apache.jena.reasoner.ValidityReport.Report;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Evaluation {

	static Logger logger = LoggerFactory.getLogger(Evaluation.class);

	public static void main(String[] args) {

		Map<String, String> ontologiesMap = new LinkedHashMap<String, String>();
		//ontologiesMap.put("http://vicodi.org/ontology#", "file:../../../ontologies/vicodi_all.owl");
		//ontologiesMap.put("http://www.ifomis.org/acgt/1.0#", "file:../../../ontologies/ACGT.owl");
		//ontologiesMap.put("http://www.co-ode.org/ontologies/galen#", "file:../../../ontologies/full-galen.owl");
		//ontologiesMap.put("http://purl.org/sig/ont/fma.owl#", "file:../../../ontologies/fma.owl");
		//ontologiesMap.put("http://ncicb.nci.nih.gov/xml/owl/EVS/Thesaurus.owl#", "file:../../../ontologies/ncit.owl");
		ontologiesMap.put("http://purl.bioontology.org/ontology/MESH/", "file:../../../ontologies/MESH.ttl");
		//ontologiesMap.put("http://purl.obolibrary.org/obo/gaz.owl#", "file:../../../ontologies/gaz.owl");
		
		Map<String, Reasoner> reasonerFactoryMap = new LinkedHashMap<>();
		reasonerFactoryMap.put("OWL Micro",  ReasonerRegistry.getOWLMicroReasoner());
		reasonerFactoryMap.put("OWL Mini",ReasonerRegistry.getOWLMiniReasoner());
		reasonerFactoryMap.put("OWL", ReasonerRegistry.getOWLReasoner());
		
		

		int RUN = 1;

		long evaluationTime = 0;

		for (String source : ontologiesMap.keySet()) {

			String filename = ontologiesMap.get(source);
			logger.info("");
			logger.info("Ontology: " + source);
			evaluationTime = 0;
			for (int i = 1; i <= RUN; i++) {
				evaluationTime += performLoadingOntologyTest(source, filename);
			}

			logger.info("Everage Loading Time of " + filename + " :" + evaluationTime / (double) RUN);

		}

		/*
		 * logger.info(""); logger.info("Evaluating Reasoner");
		 * 
		 * for(String source : ontologiesMap.keySet()) {
		 * 
		 * String filename = ontologiesMap.get(source); logger.info("Ontology: " +
		 * source);
		 * 
		 * 
		 * for(String reasonerName : reasonerFactoryMap.keySet()) { logger.info("");
		 * logger.info("Evaluation reasoner " + reasonerName); evaluationTime = 0;
		 * 
		 * for(int i = 1; i <= RUN; i++) { OntModel base =
		 * ModelFactory.createOntologyModel(OntModelSpec.OWL_MEM); OntDocumentManager dm
		 * = base.getDocumentManager(); dm.addAltEntry(source, filename);
		 * base.read(source);
		 * 
		 * Reasoner reasoner = reasonerFactoryMap.get(reasonerName);
		 * 
		 * evaluationTime += performValidationTest(base, reasoner); }
		 * 
		 * logger.info(reasonerName + " Everage Evaluation Time: " +
		 * evaluationTime/(double)RUN); }
		 * 
		 * }
		 */
	}

	public static long performLoadingOntologyTest(String source, String filename) {
		OntModel base = ModelFactory.createOntologyModel(OntModelSpec.OWL_MEM);
		OntDocumentManager dm = base.getDocumentManager();
		dm.addAltEntry(source, filename);

		long startTime = System.currentTimeMillis();
		if(filename.endsWith("ttl"))
			base.read(source, "TTL");
		else
			base.read(source);
		long endTime = System.currentTimeMillis();
		
		
		try {

			logger.info("Writing mesh file");
			base.write(new FileWriter("../../../ontologies/MESH.owl"));
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
		
		int numClassses = base.listNamedClasses().toList().size();
		int numIndividual = base.listIndividuals().toList().size();

		System.out.println("Number of Classes " + numClassses);
		System.out.println("Number of Individual " + numIndividual);

		logger.info("Loading takes " + (endTime - startTime) + " ms");

		return endTime - startTime;
	}

	public static long performValidationTest(OntModel base, Reasoner reasoner) {

		InfModel infmodel = ModelFactory.createInfModel(reasoner, base);
		
		infmodel.prepare();

	
		logger.info("Validating the ontology");
		long startTime = System.currentTimeMillis();
		ValidityReport validateReport = infmodel.validate();
		long endTime = System.currentTimeMillis();

		logger.info("Validation takes " + (endTime - startTime) + " ms");

		if (validateReport.isValid()) {
			logger.info("The Model is valid");
		} else
			logger.info("The Model is not valid");

		Iterator<Report> reports = validateReport.getReports();
		while (reports.hasNext())
			logger.info(reports.next().toString());

		return endTime - startTime;
	}

}
